### PR TITLE
Feature/us 14176

### PR DESCRIPTION
**Pull request by jhuie-zenoss**

### CODE REVIEW

File: container/endpoint.go [hunk 1]
```diff
@@ -692,15 +703,52 @@ func (c *Controller) registerExportedEndpoints() error {
...
+							glog.V(4).Infof("checking instance id of %#v equal %v", vep, c.options.Service.InstanceID)
+							if strconv.Itoa(vep.InstanceID) == c.options.Service.InstanceID {
+								glog.V(1).Infof("Deleting stale port registration for %v at %v ", portStr, path)
+								conn.Delete(path)
+							}
+						}
+					}
+				}
+
+				// TODO: avoid set if item already exist with data we want
```
Comment (smousa) [inline container/endpoint.go]: Is this TODO still applicable?

Comment (jhuie-zenoss) [inline container/endpoint.go]: Yes, the TODO is still applicable.  If an existing port is modified, then we do not check for equality and skip the update.  We remove the old node and replace it with an identical one.

**Comment by jhuie-zenoss** [comment]
Retest this, please.

**Comment by jhuie-zenoss** [comment]
Retest this, please.

**Pull request merged**